test/stack_isolation_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023, Pulumi Corporation.  All rights reserved.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pulumi/pulumi-kubernetes-operator/pkg/apis/pulumi/shared"
+	pulumiv1 "github.com/pulumi/pulumi-kubernetes-operator/pkg/apis/pulumi/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+const buildDirectoryPrefix = "pulumi-working"
+
+var _ = Describe("Stack isolation", func() {
+	var (
+		tmpDir             string
+		gitDir, backendDir string
```

Comment:
nit: We could combine these lines:
```suggestion
		tmpDir, gitDir, backendDir string
```